public/app/features/sandbox/TestStuffPage.tsx

```diff
@@ -82,6 +88,102 @@ export const TestStuffPage = () => {
                     timeZone="browser"
                   />
                   <Table data={data.series[0]} width={width} height={300} />
+                  <PanelRenderer
+                    pluginId="barchart"
+                    title="Hello"
+                    width={width}
+                    height={300}
+                    data={data}
+                    options={{}}
+                    fieldConfig={{ defaults: {}, overrides: [] }}
+                    timeZone="browser"
+                  >
+                    {(
```

Comment:
interisting -- but a bit scary since it now has totally implicit expectations on what the child object looks like.

Thoughts on how to best support this @mckn ?

Reply:
I totally see the need for this but agree that the approach of this PR is a bit scary. I think we should pass some kind of render prop for tooltips for this use case.

There have been a couple of other scenarios where we want to have panel specific properties being passed depending on what type of panel you are trying to render. I was thinking about introducing some kind of builder that will return a typed PanelRenderer depending on the pluginId.

Let me see if I can throw something together to show case this later this week.